CONTRIBUTING.md

```diff
@@ -0,0 +1,138 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue with the owners of this repository before making a change. 
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Pull Request Process
+
+1. Ensure any code or file you have created during your dev process are really needed before push it.
+2. Update the README.md with details of changes if needed.
+3. When your PR is submitted, owners and maintainers will review the PR and comment or accept it.
+
+## Tools we use to build
+- JDK 1.8+
+- Gradle 6+ (use ./gradlew[.bat] everytime !!!)
+- Kotlin 1.3+
+- Github actions (see the .github directory)
+
+## Action creation
+
+The action creation is quite simple.  
+You need to create a new Kotlin project and just add the 
+`implementation(project("io.saagie.updatarium:core:{latest-version}"))`
+
+Now you can create your own implementation of `Action`.  
+We use `com.autodsl.annotation.AutoDsl` annotation to transform our class in a simple DSL, but you can create you own DSL without.
+
+`Action` interface contains only one function : 
+
+```$kotlin
+abstract fun execute()
+```
+
+in your `execute` function you can implement your own logical. You can see existing `engine-XXX` to see some examples. 
+
+## Persist Engine creation
+
+The persist engine creation is like action creation:quite simple too.  
+
+You need to create a new Kotlin project and just add the 
+`implementation(project("io.saagie.updatarium:core:{latest-version}"))`
+
+Now you can create your own implementation of `PersistEngine`.  
+
+
+`PersistEngine` interface contains four functions : 
+
+```$kotlin
+   abstract fun checkConnection()
+   abstract fun notAlreadyExecuted(changeSetId: String): Boolean
+   abstract fun lock(changeSet: ChangeSet)
+   abstract fun unlock(changeSet: ChangeSet,status: Status)
+```
+
+Let's see in details each function : 
+
+- `checkConnection()` is call to check before start the changelog execution to assure the persistence is ready (check a database connection for exemple). This function return nothing, you have to throw an exception is the persistence is not ready, it will stop the process.
+
+- `notAlreadyExecuted(changeSetId: String)` is call at the changeset execution beginning. You have to check in your persistence engine (database ? file ? memory ? ) if you have previously execute this changesetId. Return true to allow this changeset execution, false otherwise.
```

Comment:
- `notAlreadyExecuted(changeSetId: String)` is called at the changeset execution beginning. You have to check in your persistence engine (database ? file ? memory ? ) if you have previously executed this changesetId. Return true to allow this changeset execution, false otherwise.